src/clients/crux-wallet-client.ts

```diff
@@ -0,0 +1,210 @@
+// Importing packages
+import { CruxAssetTranslator, IPutAddressMapFailures, IPutAddressMapSuccess, IResolvedClientAssetMap } from "../core/entities/crux-asset-translator";
+import { CruxDomain } from "../core/entities/crux-domain";
+import { CruxSpec } from "../core/entities/crux-spec";
+import { CruxUser, IAddress, IAddressMapping } from "../core/entities/crux-user";
+import { ICruxBlockstackInfrastructure } from "../core/interfaces";
+import { ICruxUserRepository } from "../core/interfaces/crux-user-repository";
+import { IKeyManager } from "../core/interfaces/key-manager";
+import { ICruxIDState, setCacheStorage } from "../index";
+import { BasicKeyManager } from "../infrastructure/implementations/basic-key-manager";
+import { BlockstackCruxDomainRepository } from "../infrastructure/implementations/blockstack-crux-domain-repository";
+import { BlockstackCruxUserRepository } from "../infrastructure/implementations/blockstack-crux-user-repository";
+import { BaseError, CruxClientError, ErrorHelper, PackageErrorCode } from "../packages/error";
+import { CruxDomainId, CruxId, InputIDComponents } from "../packages/identity-utils";
+import { InMemStorage } from "../packages/inmem-storage";
+import { StorageService } from "../packages/storage";
+import { cloneValue } from "../packages/utils";
+
+export const throwCruxClientError = (target: any, prop: any, descriptor?: { value?: any; }): any => {
+    let fn: any;
+    let patchedFn: any;
+    if (descriptor) {
+        fn = descriptor.value;
+    }
+    return {
+        configurable: true,
+        enumerable: true,
+        get() {
+            if (!patchedFn) {
+                patchedFn = async (...params: any[]) => {
+                    try {
+                        return await fn.call(this, ...params);
+                    } catch (error) {
+                        throw CruxClientError.fromError(error);
+                    }
+                };
+            }
+            return patchedFn;
+        },
+        set(newFn: any) {
+            patchedFn = undefined;
+            fn = newFn;
+        },
+    };
+};
+export interface ICruxWalletClientOptions {
+    privateKey?: string;
+    blockstackInfrastructure?: ICruxBlockstackInfrastructure;
+    cacheStorage?: StorageService;
+    walletClientName: string;
+}
+
+export class CruxWalletClient {
+    public walletClientName: string;
+    private cruxBlockstackInfrastructure: ICruxBlockstackInfrastructure;
+    private _initPromise: Promise<void>;
+    private _cruxUser?: CruxUser;
+    private cruxDomain?: CruxDomain;
+    private _cruxUserRepository: ICruxUserRepository;
+    private _cruxAssetTranslator?: CruxAssetTranslator;
+    private _keyManager?: IKeyManager;
+    private resolvedClientAssetMapping?: IResolvedClientAssetMap;
+    private cacheStorage?: StorageService;
+
+    constructor(options: ICruxWalletClientOptions) {
+        setCacheStorage(options.cacheStorage || new InMemStorage());
```

Comment:
you need to skip this step when you are handling cache to be handled differently per instance of client

Reply:
Replaced with `this.cacheStorage = options.cacheStorage || new InMemStorage();` as per suggestion